oar-lps/libs/oarlps/src/lib/landing/data-files/data-files.component.ts

```diff
@@ -830,31 +848,64 @@ export class DataFilesComponent implements OnInit, OnChanges {
      * Open url in a new tab
      */
     openFileManager() {
-        window.open(this.fileManagerUrl);
+        if(this.fileManagerUrl)
+            window.open(this.fileManagerUrl, 'blank');
     }
 
     /**
      * Reload data files
      */
     reloadFiles() {
+        this.refreshFilesIcon = "faa faa-spinner faa-spin icon-white";
         this.mdupdsvc.loadDataFiles().subscribe( data => {
-            let dataFiles: NerdmComp[] = data as NerdmComp[];
-            this.record['components'] = dataFiles;
-            this.buildTree(this.record['components']);
-
-            // var postMessage: any = {};
-            // postMessage[this.fieldName] = JSON.parse(JSON.stringify(this.record[this.fieldName]));
-            // console.log('postMessage', postMessage);
-
-            // // Update backend
-            // this.mdupdsvc.update(this.fieldName, postMessage).then((updateSuccess) => {
-            //     console.log("###DBG  update sent; success: "+updateSuccess.toString());
-            //     if (updateSuccess)
-            //         this.notificationService.showSuccessWithTimeout("Data files updated.", "", 3000);
-            //     else
-            //         console.error("acknowledge description update failure");
-            // });
-
-        })
+            this.mdupdsvc.loadDraft(true).subscribe({
+                next: (md) => 
+                {
+                    if(md)
+                    {
+                        this.mdupdsvc.setOriginalMetadata(md as NerdmRes);
+                        this.mdupdsvc.checkUpdatedFields(md as NerdmRes);
+
+                        this.record['components'] = JSON.parse(JSON.stringify(md['components']));
```

Comment:
This line throws an exception when `md['components']` is undefined (i.e. when `md` does not have this property).  Note that the `components` property is optional in NERDm.  In particular, this happens when the record has just been created.  Clicking the "Refresh Files" button is pressed when just after creation, before any files have been put into the file manager space.  

Consider something like this:
```javascript
                        if (md['components']) 
                            this.record['components'] = JSON.parse(JSON.stringify(md['components']));
                        else
                            this.record['components'] = []
```

Reply:
Sure. Will update soon...